ledger/catchupaccessor.go

```diff
@@ -391,34 +391,184 @@ func (c *CatchpointCatchupAccessorImpl) processStagingBalances(ctx context.Conte
 	return err
 }
 
-// EvictAsNeeded calls Evict on the cachedTrie periodically, or once we're done updating the trie.
-func (progress *CatchpointCatchupAccessorProgress) EvictAsNeeded(balancesCount uint64) (err error) {
-	if progress.cachedTrie == nil {
-		return nil
+// BuildMerkleTrie would process the catchpointpendinghashes and insert all the items in it into the merkle trie
+func (c *CatchpointCatchupAccessorImpl) BuildMerkleTrie(ctx context.Context, progressUpdates func(uint64)) (err error) {
+	wdb := c.ledger.trackerDB().wdb
+	rdb := c.ledger.trackerDB().rdb
+	err = wdb.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
+		// creating the index can take a while, so ensure we don't generate false alerts for no good reason.
+		db.ResetTransactionWarnDeadline(ctx, tx, time.Now().Add(120*time.Second))
+		return createCatchpointStagingHashesIndex(ctx, tx)
```

Comment:
Ah, yes.. that's was my initial implementation. However, realize that `processStagingBalances` is called for every chunk in the tar file. Each chunk contains 512 accounts. That means that the typical Wait would be less than 30ms. On the other hand the index creation is quite lengthy and can take 10-15 seconds.
Placing it there would "cut off" up to 30ms off the 15seconds and that wouldn't be noticeable. It would also cause a UI issue since the statistics are being updated only after the Wait.